src/TodoInput.js

```diff
@@ -1,24 +1,48 @@
-import React from 'react'
+import React, { useReducer } from 'react'
 import PropTypes from 'prop-types'
 
-const TodoInput = ({ inputValue, handleInputChange, handleAddTodo }) => {
+const todoInputReducer = (state, action) => {
+  switch (action.type) {
+    case 'CHANGE':
```

Comment:
Even if it is all in the same file, for safety, you probably want to declare these as `const` rather than strings